src/main/java/duke/taskList/TaskList.java

```diff
@@ -0,0 +1,358 @@
+
+        try {
+            if (splitCommand.length == 1) {
+                throw new DukeException("☹ OOPS!!! The description of a todo cannot be empty.");
+            }
+        } catch (DukeException e) {
+            printHorizontalLine();
+            System.out.println(e.getMessage());
+            printHorizontalLine();
+            return e.getMessage();
+        }
+        strToPrint += ("Got it, I've added this duke.task.\n");
+        String description = command.substring(5);
+        Task todoTask = new Todo(description);
+        try {
+            FileWriter myWriter = new FileWriter("./src/main/data/duke.txt", true);
+            myWriter.write("T | 0 | " + description + "\n");
+            myWriter.close();
+        } catch (IOException e) {
+            System.out.println("An error occurred.");
+            e.printStackTrace();
+            return "An error occurred";
+        }
+
+        list.add(todoTask);
+        strToPrint += (todoTask) + "\n";
+        return strToPrint += ("Now you have " + list.size() + " tasks in the list.");
+    }
+
+    /**
+     * Add a deadline into the list.
+     * @param command Command specifying what deadline to add.
+     */
+    public String addDeadline(String command) {
+        String[] splitCommand = command.split(" ");
+
+        String strToPrint = "";
+
+        try {
+            boolean hasBy = false;
+
+            if (splitCommand.length == 1) {
+                throw new DukeException("duke.task.Deadline Argument Empty");
+            }
+
+            for (int i = 0; i < splitCommand.length; i++) {
+                if (splitCommand[i].equals("/by")) {
+                    hasBy = true;
+                    System.out.println(i);
+                    if (i == 1) {
+                        throw new DukeException("There is no description");
+                    }
+                    if (i == splitCommand.length - 1) {
+                        throw new DukeException("Please insert the deadline");
+                    }
+                }
+            }
+
+            if (hasBy == false) {
+                throw new DukeException("Please insert deadline is /by when");
+            }
+
+        } catch (DukeException e) {
+            System.out.println(e.getMessage());
+            return e.getMessage();
+        }
+        strToPrint += ("Got it, I've added this deadline.\n");
+        String currStr = command.substring(9);
+        int dateIndex = currStr.indexOf("/");
+        String date = currStr.substring(dateIndex + 4);
+        String description = currStr.substring(0, dateIndex);
+        Task deadlineTask;
+        try {
+            LocalDate d1 = LocalDate.parse(date);
+            deadlineTask = new Deadline(description, date, d1);
+        } catch (DateTimeParseException e) {
+            deadlineTask = new Deadline(description, date);;
+        }
+
+        try {
+            FileWriter myWriter = new FileWriter("./src/main/data/duke.txt", true);
```

Comment:
Is it possible to store the address of the input/output file as a constant somewhere instead of hardcoding them every time? It might be easier for you to edit it as well if the file location changes afterwards

```suggestion
            FileWriter myWriter = new FileWriter(FILE_PATH, true);
```